docs/src/gs/customize-worker-node.md

```diff
@@ -0,0 +1,77 @@
+# Customizing worker nodes
+## Configure user managed boot volume encryption
+Use the following configuration in OCIMachineTemplate to use a customer
+managed boot volume encryption key
+```yaml
+kind: OCIMachineTemplate
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+spec:
+  template:
+    spec:
+      instanceSourceViaImageConfig:
+        kmsKeyId: <kms-key-id>
+```
+
+## Configure shielded instances
+Use the following configuration in OCIMachineTemplate to create [shielded instances][shielded_instances].
+Below example is for an AMD based VM.
+
+```yaml
+kind: OCIMachineTemplate
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+spec:
+  template:
+    spec:
+      platformConfig:
+        PlatformConfigType: "AMD_VM"
+        amdVmPlatformConfig:
+          isSecureBootEnabled: true
+          isTrustedPlatformModuleEnabled: true
+          isMeasuredBootEnabled: true
+```
+
+## Configure preemptible instances
+Use the following configuration in OCIMachineTemplate to create [preemtible instances][preemptible_instances].
+
+```yaml
+kind: OCIMachineTemplate
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+spec:
+  template:
+    spec:
+      preemptibleInstanceConfig:
+        terminatePreemptionAction:
+          preserveBootVolume: false
+```
+
+## Configure capacity resevration
+Use the following configuration in OCIMachineTemplate to create [preemtible instances][preemptible_instances].
+
+```yaml
+kind: OCIMachineTemplate
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+spec:
+  template:
+    spec:
+      capacityReservationId: <capacity-reservation-id>
+```
+
+## Configure Oracle Cloud Agent plugins
+Use the following configuration in OCIMachineTemplate to configure [Oracle Cloud Agent plugins][cloud_agent_plugins].
+The example below enabled Bastion plugin.
+
+```yaml
+kind: OCIMachineTemplate
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+spec:
+  template:
+    spec:
+      agentConfig:
+        pluginsConfigs:
+          - name: "Bastion"
+            desiredState: "ENABLED"
+```
```

Comment:
So this mean it will enable bastion support?

Reply:
this will enable bastion plugin, which is off by default